rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+- Ensuring that elements fit within a page is not enough. For a responsive design to be successful, the design must also be usable at all screen resolutions and sizes.
+
+### Pros and Cons of Responsive Design
+
+| Pros                          | Cons                                                                                                                                                                                                                                                                                         |
+| ----------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abundance of templates to use | Elements can move around fluidly                                                                                                                                                                                                                                                             |
+| Often easier to implement     | Longer mobile download times.                                                                                                                                                                                                                                                                |
+|                               | Tons of `@media` queries often leads to a maintenance overhead in code                                                                                                                                                                                                                       |
+|                               | Images and other high bandwidth features are simply scaled down rather than resized or eliminated outright(unless added some lines of JS which again degrades the performance and increases the mess in code), elements which may not be vital to user experience can cause poor performance |
+
+## Adaptive Design
+
+- While responsive design relies on changing the design pattern to fit the real estate available to it, **adaptive design** has multiple fixed layout sizes. When the site _detects_ the available space, it _selects_ the layout most appropriate for the screen. So, when you open a browser on the desktop, the site chooses the best layout for that desktop screen, resizing the browser has no impact on the design.
+- Adaptive works to detect the screen size and load the appropriate layout for it – generally you would design an adaptive site for six common screen widths(confirm with Saurav once on exact sizes we are doing and update):
+  - 320
+  - 480
+  - 760
+  - 960
```

Comment:
We might club `760px` & `960px` together. Need to dig deeper while researching on tablet-sized devices.

Reply:
@kamleshchandnani @saurav12 [Find the list of breakpoints we have used in the docs-website ](https://github.com/razorpay/docs/blob/master/docs-website/src/components/utils/breakpoints.ts#L1) (please ignore the breakpoints names 🙈)

These have worked well for us